include/psa/crypto_values.h

```diff
@@ -414,6 +414,20 @@
  */
 #define PSA_KEY_TYPE_DERIVE                         ((psa_key_type_t)0x1200)
 
+/** A low-entropy secret for password hashing or key derivation.
```

Comment:
(Using this thread to reply to @athoelke)

> Is the addition of PSA_KEY_TYPE_PASSWORD as a distinct type to provide a clear distinction within stored keys between arbitrary [non-secret?] data (RAW_DATA), low-entropy secrets (PASSWORD) and high-entropy secrets (DERIVE)?

Yes. Using a password with a cryptographic mechanism that requires high-entropy input is a very common mistake, and a goal of the PSA crypto API is to make such mistakes hard. It's the same reason the key agreement interface directs users to using a KDF on the raw KA output, but for passwords the consequences of misuse are much worse and so the API is stricter.